L1Trigger/L1TTrackMatch/plugins/L1TrackJetProducer.cc

```diff
@@ -200,437 +201,140 @@ void L1TrackJetProducer::produce(Event &iEvent, const EventSetup &iSetup) {
 
   if (!L1TrkPtrs_.empty()) {
     MaxZBin mzb;
+    mzb.znum = -1;
+    mzb.nclust = -1;
+    mzb.ht = -1;
+    EtaPhiBin epbins_default[phiBins_][etaBins_];  // create grid of phiBins
+
+    float phi = -1.0 * M_PI;
+    float eta = -1.0 * trkEtaMax_;
+    for (int i = 0; i < phiBins_; ++i) {
+      eta = -1.0 * trkEtaMax_;
+      for (int j = 0; j < etaBins_; ++j) {
+        epbins_default[i][j].phi = (phi + (phi + phiStep_)) / 2.0;  // phimin=phi; phimax= phimin+step
+        epbins_default[i][j].eta = (eta + (eta + etaStep_)) / 2.0;  // phimin=phi; phimax phimin+step
+        eta += etaStep_;
+      }  // for each etabin
+      phi += phiStep_;
+    }  // for each phibin (finished creating epbins)
+
+    std::vector<float> zmins, zmaxs;
+    for (int zbin = 0; zbin < zBins_; zbin++) {
+      zmins.push_back(-1.0 * trkZMax_ + zStep_ * zbin);
+      zmaxs.push_back(-1.0 * trkZMax_ + zStep_ * zbin + zStep_ * 2);
+    }
 
-    L2_cluster(L1TrkPtrs_, tdtrk_, mzb);
-    vector<Ptr<L1TTTrackType> > L1TrackAssocJet;
-    if (mzb.clusters != nullptr) {
-      for (int j = 0; j < mzb.nclust; ++j) {
-        //FILL Two Layer Jets for Jet Collection
-        if (mzb.clusters[j].pTtot <= trkPtMin_)
-          continue;  //protects against reading bad memory
-        if (mzb.clusters[j].numtracks < 1)
+    // create vectors that hold data
+    std::vector<std::vector<EtaPhiBin>> L1clusters;
+    L1clusters.reserve(phiBins_);
+    std::vector<EtaPhiBin> L2clusters;
+
+    for (unsigned int zbin = 0; zbin < zmins.size(); ++zbin) {
+      // initialize matrices
+      float zmin = zmins[zbin];
+      float zmax = zmaxs[zbin];
+      EtaPhiBin epbins[phiBins_][etaBins_];
+      std::copy(&epbins_default[0][0], &epbins_default[0][0] + phiBins_ * etaBins_, &epbins[0][0]);
+
+      //clear containers
+      L1clusters.clear();
+      L2clusters.clear();
+
+      // fill grid
+      for (unsigned int k = 0; k < L1TrkPtrs_.size(); ++k) {
+        float trkpt = L1TrkPtrs_[k]->momentum().perp();
+        float trketa = L1TrkPtrs_[k]->momentum().eta();
+        float trkphi = L1TrkPtrs_[k]->momentum().phi();
+        float trkZ = L1TrkPtrs_[k]->z0();
+        if (zmax < trkZ)
           continue;
```

Comment:
Please don't compute `trkpt` `trketa` `trkphi` if the conditions on thkZ make them useless:
```suggestion
        float trkZ = L1TrkPtrs_[k]->z0();
        if (zmax < trkZ)
          continue;
        if (zmin > trkZ)
          continue;
        if (zbin==0 && zmin == trkZ)
          continue;
        float trkpt = L1TrkPtrs_[k]->momentum().perp();
        float trketa = L1TrkPtrs_[k]->momentum().eta();
        float trkphi = L1TrkPtrs_[k]->momentum().phi();
```

Reply:
Done